client.go

```diff
@@ -815,3 +850,86 @@ func (c *TunaSessionClient) removeClosedSessions() {
 		c.Unlock()
 	}
 }
+
+func (c *TunaSessionClient) ListenUDP(addrsRe *nkngomobile.StringArray) (*udpconn.EncryptUDPConn, error) {
+	acceptAddrs, err := getAcceptAddrs(addrsRe)
+	if err != nil {
+		return nil, err
+	}
+	c.acceptAddrs = acceptAddrs
+
+	if c.wallet == nil {
+		return nil, errors.New("wallet is empty")
+	}
+
+	host, portStr, err := net.SplitHostPort(c.listeners[0].Addr().String())
+	if err != nil {
+		return nil, err
+	}
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := net.ListenUDP("udp", &net.UDPAddr{IP: net.ParseIP(host), Port: port})
+	if err != nil {
+		return nil, err
+	}
+	return udpconn.NewEncryptUDPConn(conn), nil
+}
+
+func (c *TunaSessionClient) DialUDP(remoteAddr string) (*udpconn.EncryptUDPConn, error) {
+	return c.DialUDPWithConfig(remoteAddr, nil)
+}
+
+func (c *TunaSessionClient) DialUDPWithConfig(remoteAddr string, config *nkn.DialConfig) (*udpconn.EncryptUDPConn, error) {
+	config, err := nkn.MergeDialConfig(c.config.SessionConfig, config)
+	if err != nil {
+		return nil, err
+	}
+	ctx := context.Background()
+	var cancel context.CancelFunc
+	if config.DialTimeout > 0 {
+		ctx, cancel = context.WithTimeout(ctx, time.Duration(config.DialTimeout)*time.Millisecond)
+		defer cancel()
+	}
+	pubAddrs, err := c.getPubAddrsFromRemote(ctx, remoteAddr, config)
+	if err != nil {
+		return nil, err
+	}
+	remotePublicKey, err := nkn.ClientAddrToPubKey(remoteAddr)
+	if err != nil {
+		return nil, err
+	}
+	_, err = c.getOrComputeSharedKey(remotePublicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	udpConn := new(udpconn.EncryptUDPConn)
+	addr := pubAddrs.Addrs[0]
```

Comment:
Now each item of pubAddrs might be nil. We need to use the first item that is not nil

Reply:
This might panic